### PR TITLE
Server properties fixes

### PR DESCRIPTION
**Pull request by barbaravaldez**
I separated the changes from the other PR to only include the server properties fixes.
Included the fix for https://github.com/microsoft/azuredatastudio/issues/23652
Also fixed the issue for dialog not opening properties when user is connected to localhost and a database

### CODE REVIEW

File: extensions/mssql/src/objectManagement/commands.ts [hunk 1]
```diff
@@ -111,11 +111,17 @@ async function handleObjectPropertiesDialogCommand(context: azdata.ObjectExplore
 		return;
 	}
 	try {
+		const connectedToLocalhost = context.connectionProfile.serverName === 'localhost';
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/commands.ts:114]: This seems pretty hacky - why do we need the "actual" servername? IIRC the URN doesn't even need a name for the server (since you give it a connection to the server already) - so you can just build it up as `Server/Database[@Name='...']` or whatever

File: extensions/mssql/src/objectManagement/commands.ts [hunk 2]
```diff
@@ -111,11 +111,17 @@ async function handleObjectPropertiesDialogCommand(context: azdata.ObjectExplore
...
-		const objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : undefined;
-
+		let objectType = context.nodeInfo ? context.nodeInfo.nodeType as ObjectManagement.NodeType : ObjectManagement.NodeType.Server;
+		let objectName = context.nodeInfo ? context.nodeInfo.label : context.connectionProfile.serverName;
+		let objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : undefined;
+		// TODO: fix connectionProfile not returning the actual server name when is localhost
+		if (!connectedToLocalhost && context.nodeInfo === undefined) {
+			objectType = context.connectionProfile.databaseName === '' ? ObjectManagement.NodeType.Server : ObjectManagement.NodeType.Database;
+			objectName = context.connectionProfile.databaseName === '' ? context.connectionProfile.serverName : context.connectionProfile.databaseName;
+			objectUrn = context.connectionProfile.databaseName === '' ? `Server[@Name='${context.connectionProfile.serverName}']` : `Server[@Name='${context.connectionProfile.serverName}']/Database[@Name='${context.connectionProfile.databaseName}']`;
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/commands.ts:123]: Make sure you're escaping `'`'s in these URNs whenever you have a property

File: extensions/mssql/src/objectManagement/ui/serverPropertiesDialog.ts [hunk 3]
```diff
@@ -110,8 +119,8 @@ export class ServerPropertiesDialog extends ObjectManagementDialogBase<Server, S
 		this.serviceTierInput = this.createInputBox(localizedConstants.ServiceTierText, async () => { }, this.objectInfo.serviceTier, this.options.isNewObject);
 		const serviceTierContainer = this.createLabelInputContainer(localizedConstants.ServiceTierText, this.serviceTierInput);
 
-		this.storageSpaceUsageInGBInput = this.createInputBox(localizedConstants.StorageSpaceUsageInGBText, async () => { }, this.objectInfo.storageSpaceUsageInGB.toString().concat(' GB'), this.options.isNewObject);
-		const storageSpaceUsageInGbContainer = this.createLabelInputContainer(localizedConstants.StorageSpaceUsageInGBText, this.storageSpaceUsageInGBInput);
+		this.storageSpaceUsageInMBInput = this.createInputBox(localizedConstants.StorageSpaceUsageInMBText, async () => { }, this.objectInfo.storageSpaceUsageInMB.toString().concat(' MB'), this.options.isNewObject);
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/ui/serverPropertiesDialog.ts:122]: MB should be localized - `{0} MB`

File: extensions/mssql/src/objectManagement/commands.ts (hunk 1 above)
Comment (barbaravaldez) [reply extensions/mssql/src/objectManagement/commands.ts:114]: I tried doing something similar and got and error. Let me try again just in case.

Comment (Charles-Gagnon) [reply extensions/mssql/src/objectManagement/commands.ts:114]: If not please work with @corivera and @shueybubbles to try and figure out what's going wrong there - we should at least identify what the actual bug is so we can try to address it quickly.

File: extensions/mssql/src/objectManagement/commands.ts (hunk 2 above)
Comment (barbaravaldez) [reply extensions/mssql/src/objectManagement/commands.ts:123]: Can you elaborate on this?

File: extensions/mssql/src/objectManagement/commands.ts (hunk 1 above)
Comment (barbaravaldez) [reply extensions/mssql/src/objectManagement/commands.ts]: Using `Server/Database[@Name='...']` worked! thank you!

File: extensions/mssql/src/objectManagement/commands.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/mssql/src/objectManagement/commands.ts]: If you have a database with a ' in it this breaks - e.g. `My'Database`. You end up with this URN

`Database[@Name='My'Database']`

Which isn't valid since it sees the middle ' as the "end" of the string

You need to escape quote characters whenever you quote anything - this is a similar to SQL Injection.

File: extensions/mssql/src/objectManagement/commands.ts [hunk 4]
```diff
@@ -112,9 +112,9 @@ async function handleObjectPropertiesDialogCommand(context: azdata.ObjectExplore
 	}
 	try {
 		const parentUrn = context.nodeInfo ? await getParentUrn(context) : undefined;
-		const objectType = context.nodeInfo ? context.nodeInfo.nodeType as ObjectManagement.NodeType : (context.connectionProfile.databaseName === '' ? ObjectManagement.NodeType.Server : ObjectManagement.NodeType.Database);
-		const objectName = context.nodeInfo ? context.nodeInfo.label : objectManagementLoc.PropertiesHeader;
-		const objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : undefined;
+		let objectType = context.nodeInfo ? context.nodeInfo.nodeType as ObjectManagement.NodeType : (context.connectionProfile.databaseName === '' ? ObjectManagement.NodeType.Server : ObjectManagement.NodeType.Database);
+		let objectName = context.nodeInfo ? context.nodeInfo.label : (context.connectionProfile.databaseName === '' ? context.connectionProfile.serverName : context.connectionProfile.databaseName);
+		let objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : (context.connectionProfile.databaseName === '' ? `Server[@Name='${context.connectionProfile.serverName}']` : `Server/Database[@Name='${context.connectionProfile.databaseName}']`);
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/commands.ts:117]: You shouldn't need the name for any of the URNs - just `Server` should be a valid URN as well

Comment (barbaravaldez) [reply extensions/mssql/src/objectManagement/commands.ts:117]: I need to use the Server/Database[@Name=] format to retrieve the db properties, else I get an error

Comment (Charles-Gagnon) [reply extensions/mssql/src/objectManagement/commands.ts:117]: Yeah, I was just talking about the first part of the ternary : `Server[@Name='${context.connectionProfile.serverName}']`

The other half looks fine 😄

File: extensions/mssql/src/objectManagement/commands.ts [hunk 5]
```diff
@@ -113,8 +113,8 @@ async function handleObjectPropertiesDialogCommand(context: azdata.ObjectExplore
 	try {
 		const parentUrn = context.nodeInfo ? await getParentUrn(context) : undefined;
 		const objectType = context.nodeInfo ? context.nodeInfo.nodeType as ObjectManagement.NodeType : (context.connectionProfile.databaseName === '' ? ObjectManagement.NodeType.Server : ObjectManagement.NodeType.Database);
-		const objectName = context.nodeInfo ? context.nodeInfo.label : objectManagementLoc.PropertiesHeader;
-		const objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : undefined;
+		const objectName = context.nodeInfo ? context.nodeInfo.label : (context.connectionProfile.databaseName === '' ? context.connectionProfile.serverName : context.connectionProfile.databaseName);
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/commands.ts:116]: I'd suggest doing 

`!context.connectionProfile.databaseName ? context.connectionProfile.serverName : context.connectionProfile.databaseName`

Instead since databaseName can be undefined.

**Review by Charles-Gagnon — APPROVED** [review]
Thank you for adding tests! ❤️

File: extensions/mssql/src/objectManagement/commands.ts [hunk 6]
```diff
@@ -113,8 +113,8 @@ async function handleObjectPropertiesDialogCommand(context: azdata.ObjectExplore
 	try {
 		const parentUrn = context.nodeInfo ? await getParentUrn(context) : undefined;
 		const objectType = context.nodeInfo ? context.nodeInfo.nodeType as ObjectManagement.NodeType : (context.connectionProfile.databaseName === '' ? ObjectManagement.NodeType.Server : ObjectManagement.NodeType.Database);
-		const objectName = context.nodeInfo ? context.nodeInfo.label : objectManagementLoc.PropertiesHeader;
-		const objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : undefined;
+		const objectName = context.nodeInfo ? context.nodeInfo.label : (!context.connectionProfile.databaseName ? context.connectionProfile.serverName : context.connectionProfile.databaseName);
+		const objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : (context.connectionProfile.databaseName === '' ? 'Server' : `Server/Database[@Name='${escapeSingleQuotes(context.connectionProfile.databaseName)}']`);
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/commands.ts:117]: Same comment here about using `!context.connectionProfile.databaseName`
